packages/app-desktop/gui/ConfigScreen/controls/plugins/useOnInstallHandler.test.ts

```diff
@@ -0,0 +1,97 @@
+import useOnInstallHandler from './useOnInstallHandler';
+import { renderHook } from '@testing-library/react-hooks';
+
+import PluginService, { defaultPluginSetting, PluginSetting } from '@joplin/lib/services/plugins/PluginService';
+import { ItemEvent } from './PluginBox';
+import produce from 'immer';
+
+jest.mock('@joplin/lib/services/plugins/PluginService');
+jest.mock('immer');
+
+const pluginServiceinstance = {
+	updatePluginFromRepo: jest.fn(),
+	installPluginFromRepo: jest.fn(),
+};
+
+const pluginId = 'test.plugin';
+const setInstallingPluginIds = jest.fn();
+const repoApi = jest.fn();
+const onPluginSettingsChange = jest.fn();
+const itemEvent = ({
+	item: { manifest: { id: pluginId } },
+} as ItemEvent);
```

Comment:
Isn't the same as this?

```typescript
const itemEvent:ItemEvent = {
	item: { manifest: { id: pluginId } },
}
```

If so, I would prefer this syntax (as this is what we use throughout the codebase).

Reply:
It has to be cast explicitly, because it's not really correct TS, since `ItemEvent` has more field than what we define here. But we know for sure that in this test only the `id` will be used, so `as` here overrides the TS compiler.